doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_architecture_cpu.rst

```diff
@@ -142,6 +142,23 @@ Since the nRF54H platform supports global resource sharing, where memory partiti
 
 The Secure Domain Firmware (SDFW) exposes security-related services to the Cores in the system located in local domains (like Application and Radio).
 
+.. _ug_nrf54h20_secure_domain_cracen:
+
+CRACEN hardware peripheral
+==========================
+
+The CRACEN hardware peripheral is a hardware accelerator for cryptographic operations.
+
+On nRF54H20, the CRACEN hardware peripheral is tightly integrated with the Secure Domain.
+This means that only the CPU in the Secure Domain can access the CRACEN hardware peripheral and there is no RNG peripheral available to other Cores.
+
+The hardware peripheral uses the :ref:`CRACEN driver <crypto_drivers_cracen>` to perform cryptographic operations.
+The driver implements the PSA Crypto API and relies on the IronSide Secure Element ``ug_nrf54h20_ironside_se`` firmware.
+
+The seed for CRACEN's Isolated Key Generator (IKG) is generated by a Physically Unclonable Function (PUF).
+The PUF seed is controlled by the Secure Domain ROM and neither firmware nor applications can read or modify it.
+Keys derived by this mechanism are never exposed outside CRACEN and the Secure Domain.
```

Comment:
```suggestion
Keys derived by this mechanism are never exposed outside the Secure Domain.
```

I think this sentence was too strong.

Don't you agree @degjorva ?

We put IKG derived keys into ITS don't we?

Reply:
The IKG keys themselves, (Identity key, MKEK, MEXT) are never exposed anywhere except CRACEN*, but the keys derived from the MKEK/MEXT can be exposed wherever

*At least that is true for Lumos devices, my understanding of the 54H key management is quite low